src/python/pants/reporting/streaming_workunit_handler.py

```diff
@@ -3,22 +3,22 @@
 
 import threading
 from contextlib import contextmanager
-from typing import Any, Callable, Iterator, Optional
+from typing import Any, Callable, Iterator, List, Optional
 
 
 DEFAULT_REPORT_INTERVAL_SECONDS = 10
 
 
 class StreamingWorkunitHandler:
-  def __init__(self, scheduler: Any, callback: Optional[Callable], report_interval_seconds: float = DEFAULT_REPORT_INTERVAL_SECONDS):
+  def __init__(self, scheduler: Any, callbacks: List[Callable] = [], report_interval_seconds: float = DEFAULT_REPORT_INTERVAL_SECONDS):
```

Comment:
Two things here:
1) Better to accept a `Sequence` or `Iterable` instead of `List` in function parameters when it's possible to be more flexible. `Iterable` is when you only need to do something like `for x in xs`, whereas `Sequence` is when you need things in a consistent order and use functions like  `len()`. If it can be a `Tuple` or `List`, but nothing else, use `Sequence`. If it can also include things like `Set`, then use `Iterable`.
2. It's bad practice to set the default argument to `[]` because lists are mutable. See https://docs.quantifiedcode.com/python-anti-patterns/correctness/mutable_default_value_as_argument.html.  Instead, you can default to a tuple or change to `Optional[Sequence[Callable]] = None`.